tutorials/basic_1_square/code/lib/main.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flame/palette.dart';
+import 'package:flutter/material.dart';
+import 'package:flame/game.dart';
+
+void main() {
+  runApp(GameWidget(
+    game: MyGame(),
+  ));
+}
+
+class MyGame extends Game {
+  static const int square_speed = 400;
+  late Rect squarePos;
+  int squareDirection = 1;
+
+  @override
+  Future<void> onLoad() async {
+    squarePos = Rect.fromLTWH(0, 0, 100, 100);
+  }
+
+  @override
+  void update(double dt) {
+    squarePos = squarePos.translate(square_speed * squareDirection * dt, 0);
+
+    if (squareDirection == 1 && squarePos.right > size.x) {
+      squareDirection = -1;
+    } else if (squareDirection == -1 && squarePos.left < 0) {
+      squareDirection = 1;
+    }
+  }
+
+  @override
+  void render(Canvas canvas) {
+    canvas.drawRect(squarePos, BasicPalette.white.paint);
```

Comment:
save to a static variable

Reply:
save what to a static variable?

Reply:
still no idea on what you want here

Reply:
the paint...

Reply:
I think that this shouldn't be needed, but I just checked Flame code and `.paint` is a getter that always create a new Paint, so I now understand the reasoning of your comment, but I feel what we should change Flame here, so paint is a single instance. What do you think?

Reply:
I have added this PR to address that: https://github.com/flame-engine/flame/pull/712

Reply:
that doesn't quite work because paint is mutable, so I would still recommend extracting (I see your PR is closed)